plugins/webp-uploads/hooks.php

```diff
@@ -803,3 +803,77 @@ function webp_uploads_enable_additional_mime_type_support_for_all_sizes( array $
 	return $allowed_sizes;
 }
 add_filter( 'webp_uploads_image_sizes_with_additional_mime_type_support', 'webp_uploads_enable_additional_mime_type_support_for_all_sizes' );
+
+/**
+ * Converts palette PNG images to truecolor PNG images.
+ *
+ * GD cannot convert palette-based PNG to WebP/AVIF formats, causing conversion failures.
+ * This function detects and converts palette PNG to truecolor during upload.
+ *
+ * @since n.e.x.t
+ *
+ * @param array<string, mixed>|mixed $file The uploaded file data.
+ * @return array<string, mixed> The modified file data.
+ */
+function webp_uploads_convert_palette_png_to_truecolor( $file ): array {
+	if ( ! is_array( $file ) ) {
+		$file = array();
```

Comment:
can we just return here instead of continuing?


```suggestion
		return $file;
```

Reply:
Here https://github.com/WordPress/performance/pull/2024#discussion_r2127464876 Weston suggested this:
> Since WordPress plugins can supply any kind of value in filters, this should have better safeguards to prevent a PHP fatal error:

https://github.com/WordPress/performance/pull/2024/commits/ef9a9f912619a9d2c1c5150633fc76bca92e09a1
```diff
diff --git a/plugins/webp-uploads/hooks.php b/plugins/webp-uploads/hooks.php
index 059c09fe..c9c6755f 100644
--- a/plugins/webp-uploads/hooks.php
+++ b/plugins/webp-uploads/hooks.php
@@ -812,10 +812,13 @@ add_filter( 'webp_uploads_image_sizes_with_additional_mime_type_support', 'webp_
  *
  * @since n.e.x.t
  *
- * @param array<string, mixed> $file The uploaded file data.
+ * @param array<string, mixed>|mixed $file The uploaded file data.
  * @return array<string, mixed> The modified file data.
  */
-function webp_uploads_convert_palette_png_to_truecolor( array $file ): array {
+function webp_uploads_convert_palette_png_to_truecolor( $file ): array {
+	if ( ! is_array( $file ) ) {
+		$file = array();
+	}
 	if ( ! isset( $file['tmp_name'], $file['name'] ) ) {
 		return $file;
 	}
```

I think returning an empty array makes sense here, or should we not change anything and return the `$file` as it is?

```suggestion
		return array();
```

Reply:
just return $file as it is should be fine.